metrics/ci_integrations/lib/client/buildkite/buildkite_client.dart

```diff
@@ -190,6 +190,36 @@ class BuildkiteClient with LoggerMixin {
     );
   }
 
+  /// Fetches a [BuildkiteBuild] by the given [pipelineSlug] and [buildNumber].
+  ///
+  /// Throws an [ArgumentError] if the given [pipelineSlug] or [buildNumber] is
+  /// `null`.
+  Future<InteractionResult<BuildkiteBuild>> fetchBuild(
+    String pipelineSlug,
+    int buildNumber,
+  ) {
+    ArgumentError.checkNotNull(pipelineSlug, 'pipelineSlug');
+    ArgumentError.checkNotNull(buildNumber, 'buildNumber');
```

Comment:
I'm not sure we should check this here as we don't do this in most of the methods of the client. Moreover, if we want to make this validation more developer-oriented then wouldn't it be better to replace error throwing with asserts? For production and usage purposes, I suppose it's much better when the client returns InteractionResult - especially for adapters, so each client call won't be wrapped with a try-catch. There are few points: 
- usually, the server provides more detailed errors with descriptions;
- when throwing additional errors instead of returning InteractionResult, one should consider providing an error handling to make errors clear to end-users;
- `asserts` more dev-oriented as don't work in production;
- in other places, where we're throwing `ArgumentError`, we mean that the process should be interrupted as the invalid argument makes it impossible to proceed with an algorithm - here, we still can make an API call: InteractionResult.error will be returned and the algorithm will decide what to do next.

So, according to the above, I would suggest to replace errors with asserts:
```suggestion
    assert(pipelineSlug != null);
    assert(buildNumber != null);
```
or completely remove these lines:
```suggestion
```